packages/lit-virtualizer/src/lib/lit-virtualizer.ts

```diff
@@ -37,11 +39,23 @@ export class LitVirtualizer extends LitElement {
         }
     }
 
+    /**
+     * Scroll to the specified index, placing that item at the given position
+     * in the scroll view.
+     */
+    async scrollToIndex(index: number, position: string = 'start') {
+        this._scrollTo = {index, position}
+        this.requestUpdate()
+        await this.updateComplete
+        this._scrollTo = null
```

Comment:
@graynorton Here's how I avoided that problem in LitVirtualizer. When the developer calls `scrollToIndex`, we scroll there once and then reset the internal property. That way, if the view is scrolled and then re-rendered for whatever reason, the directive will not receive a stale index.

Reply:
Yeah, this looks good.

For the `scroll` directive use case, I think we should probably just point out in the docs that you want to make sure you don't re-render with a stale value and either show a simple example or suggest some reasonable ways to avoid doing so.

Reply:
Ok! I'll push the documentation to this branch.